app-backend/db/src/test/scala/com/azavea/rf/database/DataSourceDaoSpec.scala

```diff
@@ -12,38 +14,103 @@ import cats.effect.IO
-    } yield dsOut
-
-    val result = transaction.transact(xa).unsafeRunSync
-    result.name shouldBe testName
+class DatasourceDaoSpec extends FunSuite with Matchers with Checkers with DBTestConfig with PropTestHelpers {
+
+  test("inserting a datasource") {
+    check {
+      forAll (
+        (userCreate: User.Create, orgCreate: Organization.Create, dsCreate: Datasource.Create) => {
+          val createDsIO = for {
+            orgAndUserInsert <- insertUserAndOrg(userCreate, orgCreate)
+            (orgInsert, userInsert) = orgAndUserInsert
+            dsInsert <- fixupDatasource(dsCreate, orgInsert, userInsert)
+          } yield dsInsert
+          val createDs = createDsIO.transact(xa).unsafeRunSync
+          createDs.name == dsCreate.name
+        }
+      )
+    }
   }
 
-  test("types") { check(DatasourceDao.selectF.query[Datasource]) }
-}
+  test("getting a datasource by id") {
+    check {
+      forAll (
+        (userCreate: User.Create, orgCreate: Organization.Create, dsCreate: Datasource.Create) => {
+          val getDsIO = for {
+            orgAndUserInsert <- insertUserAndOrg(userCreate, orgCreate)
+            (orgInsert, userInsert) = orgAndUserInsert
+            dsInsert <- fixupDatasource(dsCreate, orgInsert, userInsert)
+            dsGet <- DatasourceDao.getDatasourceById(dsInsert.id, userInsert)
+          } yield dsGet
+          val getDs = getDsIO.transact(xa).unsafeRunSync
+          getDs.get.name === dsCreate.name
+        }
+      )
+    }
+  }
+
+  test("getting a datasource by id unsafely") {
+    check {
+      forAll (
+        (userCreate: User.Create, orgCreate: Organization.Create, dsCreate: Datasource.Create) => {
+          val getDsUnsafeIO = for {
+            orgAndUserInsert <- insertUserAndOrg(userCreate, orgCreate)
+            (orgInsert, userInsert) = orgAndUserInsert
+            dsInsert <- fixupDatasource(dsCreate, orgInsert, userInsert)
+            dsGetUnsafe <- DatasourceDao.unsafeGetDatasourceById(dsInsert.id, userInsert)
+          } yield dsGetUnsafe
+          val getDsUnsafe = getDsUnsafeIO.transact(xa).unsafeRunSync
+          getDsUnsafe.name === dsCreate.name
+        }
+      )
+    }
+  }
+
+  test("updating a datasource") {
+    check {
+      forAll (
+        (userCreate: User.Create, orgCreate: Organization.Create, dsCreate: Datasource.Create, dsUpdate: Datasource.Create) => {
+          val updateDsIO = for {
+            orgAndUserInsert <- insertUserAndOrg(userCreate, orgCreate)
+            (orgInsert, userInsert) = orgAndUserInsert
+            dsInsert <- fixupDatasource(dsCreate, orgInsert, userInsert)
+            dsUpdated <- fixupDatasource(dsUpdate, orgInsert, userInsert)
+            rowUpdated <- DatasourceDao.updateDatasource(dsUpdated, dsInsert.id, userInsert)
+          } yield rowUpdated
+          val updateDsRowCount = updateDsIO.transact(xa).unsafeRunSync
+          updateDsRowCount == 1
```

Comment:
In other tests, I've been making comparisons between the attributes of the datasource after the update and the attributes of `dsUpdate` -- I think that makes sense to do here as well. One case where I think it catches an error that might be difficult to diagnose otherwise is if we have two attributes that are strings and accidentally put one of them in a column instead of the other.